developer_manual/app/schema.rst

```diff
@@ -1,3 +1,95 @@
+===================
+Database Migrations
+===================
+
+.. sectionauthor:: Thomas Müller <thomas.mueller@tmit.eu>
+
+ownCloud uses `Doctrine Migrations <http://www.doctrine-project.org/projects/migrations.html>`_ to manage any kind of database changes.
+
+Any apps which wants to use this mechanism has to enable it in :file:`appinfo/info.xml` by adding
+
+.. code-block:: xml
+
+  <use-migrations>true</use-migrations>
+
+
+As soon as migrations are enabled any existing :file:`appinfo/database.xml` will be ignored on upgrade of the app.
+Instead all migration step which not yet have been executed will be executed.
+
+How to develop a migration?
+===========================
+
+1. Enable migrations by adding the xml tag to :file:`appinfo/info.xml`
+
+.. code-block:: xml
+
+  <use-migrations>true</use-migrations>
+
+2. Create a migration step
+
+.. code-block:: bash
+
+  ./occ migrations:generate app-name
+
+This will create a migration step skeleton like this:
+
+.. code-block:: php
+
+  <?php
+
+  namespace OCA\dav\Migrations;
+
+  use Doctrine\DBAL\Migrations\AbstractMigration;
+  use Doctrine\DBAL\Schema\Schema;
+
+  /**
+   * Auto-generated Migration: Please modify to your needs!
+   */
+  class Version20161130090952 extends AbstractMigration
+  {
+      /**
+       * @param Schema $schema
+       */
+      public function up(Schema $schema)
+      {
+          // this up() migration is auto-generated, please modify it to your needs
+
+      }
+
+      /**
+       * @param Schema $schema
+       */
+      public function down(Schema $schema)
+      {
+          // this down() migration is auto-generated, please modify it to your needs
+
+      }
+  }
+
+Within the up() method the schema changes are to be added by working on the `Class Schema <http://www.doctrine-project.org/api/dbal/2.5/class-Doctrine.DBAL.Schema.Schema.html>`_.
+Within the down() method the schema changes are to be reverted.
+
+3. Test your migration step by executing
```

Comment:
because all steps are executed upon installation there is no real need from my pov